src/context/auth/authStorage/AuthStorage.ts

```diff
@@ -1,82 +1,93 @@
-import { Storage } from './AuthStorage.types';
+import { Storage, TokenData } from './AuthStorage.types';
 
 const ACCESS_TOKEN_KEY = 'accessToken';
 const REFRESH_TOKEN_KEY = 'refreshToken';
 const EXPIRES_KEY = 'expires';
 
+const defaultTokenData = {
+  accessToken: null,
+  refreshToken: null,
+  expires: null,
+} satisfies TokenData;
 class AuthStorage {
-  private _accessToken: string | null = null;
-  private _refreshToken: string | null = null;
-  private _expires: number | null = null;
-  private _storage: Storage | null = null;
+  private _storage;
+  private _tokenData: TokenData;
+  private _listeners: VoidFunction[] = [];
 
   constructor(_storage: Storage) {
     try {
       this._storage = _storage;
-      this.accessToken = _storage.getItem(ACCESS_TOKEN_KEY);
-      this.refreshToken = _storage.getItem(REFRESH_TOKEN_KEY);
-      this.expires = Number(_storage.getItem(EXPIRES_KEY));
+      this._tokenData = {
+        accessToken: _storage.getItem(ACCESS_TOKEN_KEY),
+        refreshToken: _storage.getItem(REFRESH_TOKEN_KEY),
+        expires: Number(_storage.getItem(EXPIRES_KEY)),
```

Comment:
if `EXPIRES_KEY` does not exist then this line will be `Number(null)` which will be evaluated to 0, therefore the default value for expires will be 0, which is different than `expires` in the `defaultTokenData` object.

Reply:
I think we'd need to do sth like `expires: _storage.getItem(EXPIRES_KEY) ? Number(_storage.getItem(EXPIRES_KEY)) : defaultTokenData.expires,`

Reply:
Yup you are right, fixed that